### DISCUSSION TITLE
Seeking Advice on Progressing as a Full-Stack Developer

### DISCUSSION THREAD

**Discussion by ChevrierDev** [community/community#88663] [Programming Help]
### Select Topic Area

General

### Body

Hope you're all doing well! I'm a student with a knack for JavaScript, PHP, and CSS, aiming to dive deep into the world of full-stack development. I've been on a quest to spice up my skills and gain a broader understanding of the full-stack landscape.

I'd love to tap into your collective wisdom with a couple of questions:

- Any cool learning hacks or methodologies you swear by for leveling up as a full-stack developer?

- Curious to hear about those game-changing projects or hands-on experiences that really propelled your growth when you first 
  started out.

Here's the deal: despite devouring every tutorial in sight, tackling countless challenges, and feeling like I'm making progress, I've hit a plateau. It's a bit disheartening. I find myself constantly seeking solutions online and struggling to implement my own fixes without that crutch.

The frustration is real, and I'm eager to break free from this cycle. If anyone has faced a similar hurdle or has advice on how to overcome this learning plateau, I would be incredibly grateful for your guidance.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Saintkops ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by ghost** [comment]
So hey @Saintkops I am currently learning css, I too caught up in these kind of hard Cycle where I couldn't learn a particular thing.

- I actually saw a complete tutorial on it to understand what was their motive to do that particular one.
- Then if I don't understood it enough I would read the documentation of it on W3schools or MDN
- still if I didn't understand it the. I would leave it then work on small projects with the concepts I know to strengthen my prev Knowledge.
- then I would learn other easier concepts apart from it.
- so this is where you got a small break here your mind would be pleased and confident that you can learn that one particular concept.
- Then if you give another try learning it im 75% sure that you can grasp those concepts.

Hope this helped you.

**Reply by ChevrierDev** [reply]
Hey @Yuheshpandian, Thanks a bunch for sharing your learning journey with CSS! It's awesome that you're diving into the nitty-gritty of it. Learning new stuff can indeed be a bit of a rollercoaster, but your approach sounds solid!

Happy coding! 🚀

**Reply by ghost** [reply]
Your welcome ☺️.

Happy Web Development Journey 🚀🎉✨

**Comment by anirbandotdev** [comment]
Hey @Saintkops , its great to hear that you have been actively exploring JavaScript, PHP, and CSS and aspiring to become a proficient full-stack developer. I'm on a similar journey to become a full-stack developer, and I completely resonate with the challenges you're facing. It's awesome that you're actively seeking advice and looking to break through the learning plateau. One strategy that has worked for me is diving into collaborative coding environments or finding a mentor . Join communities in discord , find coding groups and discuss with them , it really helps . First work on a few pure html , css and js projects . It can be a basic static website or a simple dynamic website with decent styling . Then go for React , learn the basics of react , states , state management , hooks and all . Then after spending time on the standalone react projects then go for backend . You can find a mentor in youtube or any paid plateform for your journey . Also go for CSS frameworks like TailwindCSS . Work on diverse projects with real-world applications to deepen your understanding. Remember, the learning curve is part of the process, and each challenge is an opportunity to grow. Keep pushing forward, and feel free to share your progress or ask for insights anytime! 🚀💻

**Reply by ChevrierDev** [reply]
Hey @AnirbanNath-dev!

Thanks a ton for your encouraging words and fantastic advice! It's great to connect with someone on a similar full-stack journey. Your strategy of diving into collaborative coding environments and seeking mentors is spot on!
diving into React sounds like the next thrilling chapter. States, hooks, and all the React magic—I'm ready for it!
TailwindCSS is also on my radar now, thanks to your recommendation. Real-world projects are where the rubber meets the road, right?
Quick question: Any specific resources or YouTube mentors you found particularly helpful during your journey?
Also, since you mentioned Discord communities, do you happen to know any vibrant groups with potential mentors or just awesome devs exchanging ideas? I'm all about that collaborative spirit!
Looking forward to hearing more about your journey and any insights you might have.🚀💻

**Comment by ghost** [comment]
Just a little advice/insight to share.

"Whatever happens, whatever leaves you, whatever you loose, whatever stops your path, just keep going until you succeed. For me success is the view from the top"
